Modules/ITS/src/ITSFhrCheck.cxx

```diff
@@ -33,77 +33,68 @@ void ITSFhrCheck::configure() {}
+      result = Quality::Good;
+      auto* h = dynamic_cast<TH1D*>(mo->getObject());
+      if (h->GetMaximum() > 0) {
+        result.set(Quality::Bad);
+      }
+    } else if (mo->getName() == "General/General_Occupancy") {
+      auto* h = dynamic_cast<TH2Poly*>(mo->getObject());
+      result.addMetadata("Gen_Occu", "good");
+      for (int ibin = 0; ibin < h->GetNumberOfBins(); ++ibin) {
+        if (h->GetBinContent(ibin) == 0) {
+          continue;
+        }
+        if (h->GetBinContent(ibin) > pow(10, -5)) {
+          result.updateMetadata("Gen_Occu", "bad");
+        } else if (h->GetBinContent(ibin) > pow(10, -6) && (strcmp(result.getMetadata("Gen_Occu").c_str(), "good") == 0)) {
+          result.updateMetadata("Gen_Occu", "medium");
         }
       }
-    } else if (iter->second->getName() == "General/Noisy_Pixel") {
-      auto* h = dynamic_cast<TH2Poly*>(iter->second->getObject());
+    } else if (mo->getName() == "General/Noisy_Pixel") {
+      auto* h = dynamic_cast<TH2Poly*>(mo->getObject());
+      result.addMetadata("Noi_Pix", "good");
       for (int ibin = 0; ibin < h->GetNumberOfBins(); ++ibin) {
         if (h->GetBinContent(ibin) == 0) {
           continue;
         }
         if (ibin < 48) {
           if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[0]) > 0.0001) {
-            result = (result.getLevel() % 100) + 100;
-          } else if (h->GetBinContent(ibin) > 0.00005 && (result.getLevel() / 100 < 1)) {
-            result = (result.getLevel() % 100) + 200;
+            result.updateMetadata("Noi_Pix", "bad");
+          } else if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[0]) > 0.00005 && strcmp(result.getMetadata("Noi_Pix").c_str(), "good") == 0) {
+            result.updateMetadata("Noi_Pix", "medium");
           }
         } else if (ibin < 102) {
           if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[1]) > 0.0001) {
-            result = (result.getLevel() % 100) + 100;
-          } else if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[1]) > 0.00005 && (result.getLevel() / 100 < 1)) {
-            result = (result.getLevel() % 100) + 200;
+            result.updateMetadata("Noi_Pix", "bad");
+          } else if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[1]) > 0.00005 && strcmp(result.getMetadata("Noi_Pix").c_str(), "good") == 0) {
+            result.updateMetadata("Noi_Pix", "medium");
           }
         } else {
           if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[2]) > 0.0001) {
-            result = (result.getLevel() % 100) + 100;
-          } else if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[2]) > 0.00005 && (result.getLevel() / 100 < 1)) {
-            result = (result.getLevel() % 100) + 200;
+            result.updateMetadata("Noi_Pix", "bad");
+          } else if ((h->GetBinContent(ibin) / (double)mNPixelPerStave[2]) > 0.00005 && strcmp(result.getMetadata("Noi_Pix").c_str(), "good") == 0) {
```

Comment:
do you really need the `strcmp` in this `else if` condition? You enter this `else if` only if the fraction of noisy pixels is < 0.0001. I might miss something, please let me know.